api/v1alpha1/clusterinstance_types.go

```diff
@@ -406,6 +436,27 @@ type ManifestReference struct {
 	Message string `json:"message,omitempty"`
 }
 
+type ReinstallHistory struct {
+	// Observed reinstall generation.
+	// + required
+	ObservedGeneration string `json:"observedGeneration"`
+	// Date/time of the reinstallation.
+	// +required
+	Timestamp metav1.Time `json:"timestamp"`
+	// Difference between last and current ClusterInstnace spec fields as a JSON string.
```

Comment:
This is interesting. How large do we expect this to be?

Reply:
This should be small as we confine the permissible changes to the ClusterInstance spec fields.